pkg/server/container_execsync.go

```diff
@@ -131,50 +131,22 @@ func (c *criContainerdService) execInContainer(ctx context.Context, id string, o
 		}
 	})
 
-	// Get containerd event client first, so that we won't miss any events.
-	// TODO(random-liu): Add filter to only subscribe events of the exec process.
-	// TODO(random-liu): Use `Wait` after is fixed. (containerd#1279, containerd#1287)
-	cancellable, cancel := context.WithCancel(ctx)
-	eventstream, err := c.eventService.Subscribe(cancellable, &events.SubscribeRequest{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to subscribe event stream: %v", err)
-	}
-	defer cancel()
-
+	resCh := make(chan execResult, 1)
+	go func() {
```

Comment:
You are correct, I even found a containerd bug when trying to address this https://github.com/containerd/containerd/issues/1376.